Improve inline var parsing, add a simple unit test of inline var parsing

TysonAndre opened the pull request:
Allow multiple var annotations in a single inline doc comment.

Make this behave similarly to Comment::parameterFromCommentLine but be more specific for inline variables, where names and types must be provided to this plugin.

Also allow parsing unions of types(i.e. `|`) and various other type syntaxes, even generic arrays of types (`@var T[] $x`).

TysonAndre committed:
WIP - Support union types

TysonAndre committed:
Write a simple test suite, fix bugs in my changes

bobvandevijver commented on Helper/NamespaceChecker.php:
```diff
@@ -5,95 +5,76 @@
-        // See https://github.com/phan/phan/pull/1467
-        $context->getNamespaceMapFor(USE_NORMAL, $name);
-        return true;
-      }
```
The `NamespaceChecker` is also used for the `SymfonyAnnotationVisitor`, which is now no longer working as expected, as it throws an error for every annotation it can find. 

For example, this snippet
```php
<?php

namespace Idb\EventBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use JMS\Serializer\Annotation as JMS;

/**
 * Weather
 *
 * @ORM\Table()
 * @ORM\Entity
 * @JMS\ExclusionPolicy("all")
 */
class Weather
{
  /**
   * @var integer
   *
   * @ORM\Column(name="id", type="integer")
   * @ORM\Id
   * @ORM\GeneratedValue(strategy="AUTO")
   * @JMS\Expose()
   */
  private $id;

  /**
   * @var string
   *
   * @ORM\Column(name="code", type="string", length=20)
   * @JMS\Expose()
   */
  private $code;
}
```
will now throw
```
src/Idb/EventBundle/Entity/Weather.php:18 AnnotationNotImported The classlike/namespace \Doctrine\ORM\Mapping was never imported (generated by DrensoAnnotation plugin)
src/Idb/EventBundle/Entity/Weather.php:18 AnnotationNotImported The classlike/namespace \JMS\Serializer\Annotation was never imported (generated by DrensoAnnotation plugin)
src/Idb/EventBundle/Entity/Weather.php:32 AnnotationNotImported The classlike/namespace \Doctrine\ORM\Mapping was never imported (generated by DrensoAnnotation plugin)
src/Idb/EventBundle/Entity/Weather.php:32 AnnotationNotImported The classlike/namespace \JMS\Serializer\Annotation was never imported (generated by DrensoAnnotation plugin)
src/Idb/EventBundle/Entity/Weather.php:40 AnnotationNotImported The classlike/namespace \Doctrine\ORM\Mapping was never imported (generated by DrensoAnnotation plugin)
src/Idb/EventBundle/Entity/Weather.php:40 AnnotationNotImported The classlike/namespace \JMS\Serializer\Annotation was never imported (generated by DrensoAnnotation plugin)
```

I believe this is due to the removal of this part (`getNamespaceMapFor`). Can you confirm?

TysonAndre replied:
https://github.com/schmittjoh/serializer/blob/master/src/JMS/Serializer/Annotation/Expose.php

 preg_match_all('/\s*\*\s*\@([A-Z][a-zA-Z]+)[\\\(]?/', $docComment, $matches);

If you had a regex that would also match the entire annotation (e.g. `ORM\Column(` and `ORM\Id`), then my changes would probably work.)

I'm refraining from modifying the symfony plugin to check for classes instead of namespaces in the latest version of PR, since I don't know what sort of false positives you would encounter, and I'm not familiar with symfony.

Anything with at least one `\`, or anything ending with `(` might be symfony

bobvandevijver reviewed (changes requested):
Thanks for your enormous contribution! I do have a single question though.

TysonAndre committed:
Address review comments, don't break the ORM plugin

InlineVarPlugin should just use UnionType::fromStringInContext instead.

Meanwhile, the SymfonyAnnotationPlugin works as previously written

- Remove checkPlugin for now, that can be restored if it gets used again

bobvandevijver commented:
Closing in favour of #3.
